### PR TITLE
perf(neon): do not use the spread operator for building lists

### PR DESCRIPTION
**Pull request by Leptopoda**
Fixes: #696

### CODE REVIEW

**Comment by provokateurin** [comment]
Draft is on purpose?

**Comment by Leptopoda** [comment]
This time yes.
I made the NeonProgressIndicator conditional and wanted to remove the `visible` property in a followup but noticed that this would cause flickering when it gets removed.
I already reverted most of them locally and we'll need to wait for #800 to properly change this.

**Comment by Leptopoda** [comment]
Oh totally forgot to mark it as ready for review :woman_facepalming:

File: packages/neon/neon/lib/src/pages/login.dart [hunk 1]
```diff
@@ -67,56 +67,55 @@ class _LoginPageState extends State<LoginPage> {
...
+                      child: Text(NeonLocalizations.of(context).loginWorksWith),
                     ),
-                    Text(NeonLocalizations.of(context).loginWorksWith),
-                    const SizedBox(
-                      height: 10,
-                    ),
-                    Semantics(
-                      label: NeonLocalizations.of(context).nextcloud,
-                      child: const NextcloudLogo(),
+                  if (branding.showLoginWithNextcloud)
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/pages/login.dart:75]: Duplicate condition, better use a generator function for the whole thing

File: packages/neon/neon_news/lib/dialogs/add_feed.dart
Comment (provokateurin) [inline packages/neon/neon_news/lib/dialogs/add_feed.dart:1]: Duplicate condition -> generator function

File: packages/neon/neon_news/lib/pages/article.dart
Comment (provokateurin) [inline packages/neon/neon_news/lib/pages/article.dart:1]: Duplicate condition -> generator function

File: packages/neon/neon_notes/lib/dialogs/create_note.dart
Comment (provokateurin) [inline packages/neon/neon_notes/lib/dialogs/create_note.dart:1]: Duplicate condition -> generator function

File: packages/neon/neon_notes/lib/widgets/notes_view.dart
Comment (provokateurin) [inline packages/neon/neon_notes/lib/widgets/notes_view.dart:1]: Duplicate condition -> generator function

**Review by provokateurin — CHANGES_REQUESTED** [review]
Oops I forgot about this PR

**Comment by Leptopoda** [comment]
Thanks for the feedback.
I'll touch this PR up after all the dynamite work I'm currently working on.

**Comment by Leptopoda** [comment]
Closing this one.
There is no point in rebasing it.

**Pull request closed without merging**